### PR TITLE
[ROCm] improve use of ROCm libraries, enable more tests, small fixes

### PR DESCRIPTION
**Pull request by iotamudelta**
* some small leftovers from the last PR review
* enable more unit test sets for CI
* replace use of hcRNG w/ rocRAND (docker image was already updated w/ newer rocRAND)
* use rocBLAS instead of hipBLAS to allow convergence w/ Caffe2
* use strided_batched gemm interface also from the batched internal interface
* re-enable Dropout.cu as we now have philox w/ rocRAND

### CODE REVIEW

File: aten/src/THC/THCBlas.cu [hunk 1]
```diff
@@ -374,6 +378,25 @@ void THCudaBlas_HgemmStridedBatched(THCState *state, char transa, char transb, i
...
+void THCudaBlas_SgemmBatched(THCState *state, char transa, char transb, int64_t m, int64_t n, int64_t k,
+                             float alpha, const float *a[], int64_t lda, const float *b[], int64_t ldb,
+                             float beta, float *c[], int64_t ldc, int64_t batchCount)
+{
+  if( (m >= INT_MAX) || (n >= INT_MAX) || (k >= INT_MAX) || (lda >= INT_MAX)  || (ldb >= INT_MAX) || (ldc >= INT_MAX) || (batchCount >= INT_MAX) )
+  {
+    THError("Cublas_SgemmBatched only supports m, n, k, lda, ldb, ldc, batchCount"
+            "with the bound [val] <= %d", INT_MAX);
+  }
+
```
Comment (ezyang) [inline aten/src/THC/THCBlas.cu]: Why does the ifdef span over the entire function? The signature and the error checking look the same. Just ifdef in the body.

File: aten/src/THC/THCBlas.cu [hunk 2]
```diff
@@ -421,6 +445,25 @@ void THCudaBlas_SgemmStridedBatched(THCState *state, char transa, char transb, i
 }
 #endif
 
+#ifdef __HIP_PLATFORM_HCC__
+void THCudaBlas_DgemmBatched(THCState *state, char transa, char transb, int64_t m, int64_t n, int64_t k,
+                             double alpha, const double *a[], int64_t lda, const double *b[], int64_t ldb,
+                             double beta, double *c[], int64_t ldc, int64_t batchCount)
+{
+  if( (m >= INT_MAX) || (n >= INT_MAX) || (k >= INT_MAX) || (lda >= INT_MAX)  || (ldb >= INT_MAX) || (ldc >= INT_MAX) || (batchCount >= INT_MAX) )
```
Comment (ezyang) [inline aten/src/THC/THCBlas.cu]: Same here

**Review by ezyang — APPROVED** [review]

**Review by ezyang — DISMISSED** [review]
Lint failed. This must be fixed before we can merge.

File: aten/src/ATen/CMakeLists.txt [hunk 3]
```diff
@@ -256,10 +256,10 @@ ENDIF()
 
 IF(USE_ROCM)
  ### Link in the ROCm libraries BLAS / RNG.
- FIND_LIBRARY(HIPBLAS_LIBRARY hipblas HINTS ${HIPBLAS_PATH}/lib)
- FIND_LIBRARY(HIPRNG_LIBRARY hcrng HINTS ${HIPRNG_PATH}/lib)
+ FIND_LIBRARY(ROCBLAS_LIBRARY rocblas HINTS ${ROCBLAS_PATH}/lib)
+ FIND_LIBRARY(HIPRAND_LIBRARY hiprand HINTS ${HIPRAND_PATH}/lib)
 
- list(APPEND ATen_CUDA_DEPENDENCY_LIBS ${HIPBLAS_LIBRARY} ${HIPRNG_LIBRARY})
+ list(APPEND ATen_CUDA_DEPENDENCY_LIBS ${ROCBLAS_LIBRARY} ${HIPRAND_LIBRARY})
```
Comment (bddppq) [inline aten/src/ATen/CMakeLists.txt]: Can this be merged with Caffe2_HIP_DEPENDENCY_LIBS?

Comment (iotamudelta) [reply aten/src/ATen/CMakeLists.txt]: Yes, the idea is to have these merged as a next step. @petrex and I will talk about it.

Comment (bddppq) [reply aten/src/ATen/CMakeLists.txt]: @iotamudelta cool thanks

**Pull request closed without merging**